File: app/src/main/java/com/roman/kubik/spivanyklicejista/presentation/main/MainContract.java
package com.roman.kubik.spivanyklicejista.presentation.main;

import com.roman.kubik.spivanyklicejista.domain.song.Song;

import java.util.List;

/**
 * Contract between {@link MainActivity} and {@link MainPresenter}
 * Created by kubik on 1/14/18.
 */

public interface MainContract {

    interface View {
        void showProgress(boolean show);
        void showError(String errorMessage);
        void onSongsFetched(List<Song> songList);
    }

    interface Presenter {
        void fetchAllSongs();
    }
}

File: app/src/main/java/com/roman/kubik/spivanyklicejista/presentation/main/MainContract.kt
package com.roman.kubik.spivanyklicejista.presentation.main

import com.roman.kubik.spivanyklicejista.domain.song.Song

interface MainContract {

    interface View {
        fun setPatrioticsCount(count: Int)
        fun setBonfiresCount(count: Int)
        fun setAbroadsCount(count: Int)
        fun setAllCount(count: Int)
        fun showError(error: Throwable)
        fun navigateToSong(song: Song)
    }

    interface Presenter {
        fun requestData()
        fun requestRandom()
        fun destroy()
    }
}